1-org/modules/centralized-logging/variables.tf

```diff
@@ -45,74 +45,55 @@ variable "logging_destination_project_id" {
   type        = string
 }
 
-
 #----------------------------- #
 # Logbucket specific variables #
 #----------------------------- #
 variable "logbucket_options" {
   description = <<EOT
 Destination LogBucket options:
-- logging_sink_name: The name of the log sink to be created.
-- logging_sink_filter: The filter to apply when exporting logs. Only log entries that match the filter are exported. Default is '' which exports all logs.
 - name: The name of the log bucket to be created and used for log entries matching the filter.
+- logging_sink_name: The name of the log sink to be created.
+- logging_sink_filter: The filter to apply when exporting logs. Only log entries that match the filter are exported. Default is "" which exports all logs.
 - location: The location of the log bucket. Default: global.
-- retention_days: (Optional) The number of days data should be retained for the log bucket. Default 30.
-
-Destination LogBucket options example:
-```
-logbucket_options = {
-  logging_sink_name   = "sk-c-logging-logbkt"
-  logging_sink_filter = ""
-  name                = "logbkt-org-logs"
-  retention_days      = "30"
-  location            = "global"
-}
-```
+- retention_days: The number of days data should be retained for the log bucket. Default 30.
 EOT
-  type        = map(any)
-  default     = null
+  type = object({
+    name                = optional(string, null)
+    logging_sink_name   = optional(string, null)
+    logging_sink_filter = optional(string, "")
+    location            = optional(string, "global")
+    retention_days      = optional(number, 30)
+  })
+  default = null
 
   validation {
-    condition     = var.logbucket_options == null ? true : !contains(keys(var.logbucket_options), "retention_days") ? true : can(tonumber(var.logbucket_options["retention_days"]))
+    condition     = var.logbucket_options == null ? true : can(tonumber(var.logbucket_options.retention_days))
```

Comment:
Since `retention_days` is  now typed as a number, I think we can remove casting and can.

Reply:
removed